### ISSUE TITLE
using the source code directly with some editing in my project

### ISSUE THREAD

**Issue by jaraheel** [darrencl/FeatureSelector.jl#5]
Hi,
I tried installing your library using Julia package manager, but that gives some compatibility issues with the DataFrames, Flux, and a few other libraries' versions I have installed. To circumvent this issue, I downloaded your source files and added them to my project with some editing. Is this ok? I would credit you in the published code and documentation for my project.

**Comment by darrencl** [comment]
Hi @jaraheel,

Thank you for using this package! I haven't worked on this for a while. May I ask if your project is a private project or you'll make it public? 

I would rather fixing t his issue. I have created a PR (#6) to fix this and will merge and do minor release once passed the CI. The project itself runs fine, but I need to fix the documenter CI. I will let you know once it's done so you can update your version of FeatureSelectors.

In the meantime, if you wish, you can use this version by:

```julia-repl
(@v1.7) pkg> add FeatureSelectors#bump-deps
```

Thanks,
Darren

**Comment by jaraheel** [comment]
Hi Darren,
My project would eventually be public. It is for academic, non-commercial purposes.

**Comment by jaraheel** [comment]
just attempted to add the library with #bump-deps, and it still conflicts with DataFrames:

ERROR: Unsatisfiable requirements detected for package DataFrames [a93c6f00]:
 DataFrames [a93c6f00] log:
 ├─possible versions are: 0.11.7-1.3.2 or uninstalled
 ├─restricted to versions 1.3.0-1 by FeatureSelectors [e0e7d8aa], leaving only versions 1.3.0-1.3.2
 │ └─FeatureSelectors [e0e7d8aa] log:
 │   ├─possible versions are: 0.1.0 or uninstalled
 │   └─FeatureSelectors [e0e7d8aa] is fixed to version 0.1.0
 └─restricted by compatibility requirements with LowRankModels [15d4e49f] to versions: [0.11.7-0.19.4, 0.21.0-0.21.8] — no versions left
   └─LowRankModels [15d4e49f] log:
     ├─possible versions are: 1.0.0-1.1.1 or uninstalled
     └─restricted to versions * by an explicit requirement, leaving only versions 1.0.0-1.1.1

**Comment by darrencl** [comment]
Hi @jaraheel,

Thank you for trying that! I think I just fixed it by adding more flexibility with caret specifier in the compat. Could you try re-adding `FeatureSelectors#bump-deps` again?

 
Below is the example of my environment in Julia 1.7. I've able to install this alongside with DataFrame 1.3.2.
```
(@v1.7) pkg> status
      Status `~/.julia/environments/v1.7/Project.toml`
  [a93c6f00] DataFrames v1.3.2
  [e0e7d8aa] FeatureSelectors v0.1.1 `https://github.com/darrencl/FeatureSelectors.jl.git#bump-deps`
  [587475ba] Flux v0.12.9
  [09f84164] HypothesisTests v0.10.6
  [ce6b1742] RDatasets v0.7.7
  [2913bbd2] StatsBase v0.33.16
```

Hope this fixes the issue!

**Comment by darrencl** [comment]
@jaraheel Seems like the CI passed. I will squash merge tomorrow then release minor version tomorrow.

The current version of `bump-deps` branch should work for you, let me know how that goes.

**Comment by jaraheel** [comment]
it still gives me the same error
pkg> add FeatureSelectors#bump-deps
    Updating git-repo `https://github.com/darrencl/FeatureSelectors.jl.git`
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package DataFrames [a93c6f00]:
 DataFrames [a93c6f00] log:
 ├─possible versions are: 0.11.7-1.3.2 or uninstalled
 ├─restricted to versions 1.3.0-1 by FeatureSelectors [e0e7d8aa], leaving only versions 1.3.0-1.3.2
 │ └─FeatureSelectors [e0e7d8aa] log:
 │   ├─possible versions are: 0.1.1 or uninstalled
 │   └─FeatureSelectors [e0e7d8aa] is fixed to version 0.1.1
 └─restricted by compatibility requirements with LowRankModels [15d4e49f] to versions: [0.11.7-0.19.4, 0.21.0-0.21.8] — no versions left
   └─LowRankModels [15d4e49f] log:
     ├─possible versions are: 1.0.0-1.1.1 or uninstalled
     └─restricted to versions * by an explicit requirement, leaving only versions 1.0.0-1.1.1

**Comment by darrencl** [comment]
Hi @jaraheel 

I think your issue is not with `FeatureSelectors.jl` anymore. `DataFrames.jl` has already released v1.* and it seems that `LowRankModels.jl` locks it to the older versions.

I just checked and found this issue (https://github.com/madeleineudell/LowRankModels.jl/issues/120). Someone has submitted a PR (https://github.com/madeleineudell/LowRankModels.jl/pull/121) to fix this. Let's hope this will get merged soon.

All in all, you should use latest released version of `DataFrames.jl`. So, what you need to do is:

* Remove `LowRankModels.jl`, i.e. from repl `]remove LowRankModels`
* Update your `DataFrames.jl`, i.e. `]up DataFrames`. Ensure you have latest version installed by checking package status `]status DataFrames`
* Add `LowRankModels.jl` from the PR commit hash `]add LowRankModels#e15afec`

Below is an example in my environment. It fails to install `LowRankModels.jl` released version because it locks DataFrames to older version (while I have newer version installed and my package require newer version, since there were some breaking changes since 0.21 —  version required by released `LowRankModels.jl`). 

```
(@v1.7) pkg> add LowRankModels
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package LowRankModels [15d4e49f]:
 LowRankModels [15d4e49f] log:
 ├─possible versions are: 1.0.0-1.1.1 or uninstalled
 ├─restricted to versions * by an explicit requirement, leaving only versions 1.0.0-1.1.1
 ├─restricted by compatibility requirements with StatsBase [2913bbd2] to versions: 1.1.0-1.1.1 or uninstalled, leaving only versions: 1.1.0-1.1.1
 │ └─StatsBase [2913bbd2] log:
 │   ├─possible versions are: 0.24.0-0.33.16 or uninstalled
 │   └─restricted to versions 0.33 by FeatureSelectors [e0e7d8aa], leaving only versions 0.33.0-0.33.16
 │     └─FeatureSelectors [e0e7d8aa] log:
 │       ├─possible versions are: 0.1.1 or uninstalled
 │       └─FeatureSelectors [e0e7d8aa] is fixed to version 0.1.1
 └─restricted by compatibility requirements with DataFrames [a93c6f00] to versions: uninstalled — no versions left
   └─DataFrames [a93c6f00] log:
     ├─possible versions are: 0.11.7-1.3.2 or uninstalled
     └─restricted to versions 1.3.0-1 by FeatureSelectors [e0e7d8aa], leaving only versions 1.3.0-1.3.2
       └─FeatureSelectors [e0e7d8aa] log: see above
```

Using that commit from PR solves the issue since it's now able to use newer `DataFrames.jl` version. It also seems to pass their CI, so it should be good to go and safe for you to use.
```
(@v1.7) pkg> add LowRankModels#e15afec
     Cloning git-repo `https://github.com/madeleineudell/LowRankModels.jl.git`
    Updating git-repo `https://github.com/madeleineudell/LowRankModels.jl.git`
   Resolving package versions...
   Installed RandomizedLinAlg ─── v0.1.0
   Installed Arpack_jll ───────── v3.8.0+0
   Installed NMF ──────────────── v0.5.3
   Installed NLSolversBase ────── v7.8.2
   Installed FiniteDiff ───────── v2.11.0
   Installed NonNegLeastSquares ─ v0.4.0
   Installed Parameters ───────── v0.12.3
   Installed Arpack ───────────── v0.5.0
   Installed Optim ────────────── v1.6.2
  Downloaded artifact: Arpack
    Updating `~/.julia/environments/v1.7/Project.toml`
  [15d4e49f] + LowRankModels v1.1.1 `https://github.com/madeleineudell/LowRankModels.jl.git#e15afec`
    Updating `~/.julia/environments/v1.7/Manifest.toml`
  [7d9fca2a] + Arpack v0.5.0
  [6a86dc24] + FiniteDiff v2.11.0
  [d3d80556] + LineSearches v7.1.1
  [15d4e49f] + LowRankModels v1.1.1 `https://github.com/madeleineudell/LowRankModels.jl.git#e15afec`
  [d41bc354] + NLSolversBase v7.8.2
  [6ef6ca0d] + NMF v0.5.3
  [b7351bd1] + NonNegLeastSquares v0.4.0
  [429524aa] + Optim v1.6.2
  [d96e819e] + Parameters v0.12.3
  [85a6dd25] + PositiveFactorizations v0.2.4
  [0448d7d9] + RandomizedLinAlg v0.1.0
  [6e75b9c4] + ScikitLearnBase v0.5.0
  [3a884ed6] + UnPack v1.0.2
  [68821587] + Arpack_jll v3.8.0+0
Precompiling project...
  14 dependencies successfully precompiled in 12 seconds (118 already precompiled)

```

Hope this helps!

**Comment by jaraheel** [comment]
it worked! Thank you very much. Time to test :)
